### PR TITLE
First draft of the migration guide

### PR DESCRIPTION
**Pull request by jleibs**
### What
First draft at a structure for the migration guide. Assume there will be more incremental cleanups.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3638) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3638)
- [Docs preview](https://rerun.io/preview/d9225021ae0d7327e8fda9bda9a0064267f61883/docs) 
- [Examples preview](https://rerun.io/preview/d9225021ae0d7327e8fda9bda9a0064267f61883/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: docs/content/reference/migration-0-9.md [hunk 1]
```diff
@@ -3,4 +3,238 @@ title: Migration to 0.9
...
+All of the previous `log_*` functions have been marked as deprecated and will be removed in `0.10`. We have done our
+best to keep these functions working as thin wrappers on top of the new logging APIs, though there may be subtle
+behavioral differences.
+
+## The log module has become the log function
+This is one area where we were forced to make breaking changes.  Rerun previously had an internal `log` module where the
+assorted log-functions and helper classes were implemented. In general, these symbols were all re-exported to the
+top-level `rerun` namespace.  However, in some cases these fully-qualified paths were used for imports. Because
+`rerun.log` is now a function rather than a module, any such imports will result in an import error. Look for the
+corresponding symbol in the top-level `rerun` namespace instead.
```
Comment (emilk) [inline docs/content/reference/migration-0-9.md:38]:
```suggestion
corresponding symbol in the top-level `rerun` namespace instead.
For instance: `rr.log.text.LoggingHandler` → `rr.LoggingHandler`.
```

File: docs/content/reference/migration-0-9.md [hunk 2]
```diff
@@ -3,4 +3,238 @@ title: Migration to 0.9
...
+
+Python docs: [DepthImage.__init__](https://ref.rerun.io/docs/python/HEAD/common/images/#rerun.DepthImage.__init__)
+
+Notes:
+ * `image` has become `data`
+
+### `log_disconnected_space`
+Replace with [DisconnectedSpace](data_types/archetypes/disconnected_space.md)
+
+Python docs: [DisconnectedSpace.__init__](https://ref.rerun.io/docs/python/HEAD/common/transforms_and_coordinate_systems/#rerun.DisconnectedSpace.__init__)
```
Comment (emilk) [inline docs/content/reference/migration-0-9.md:81]: I appreciate you linking directly to the `init` method, but having that _visible_ is distracting imho. Therefore I suggest we strip it everywhere:

```suggestion
Python docs: [DisconnectedSpace](https://ref.rerun.io/docs/python/HEAD/common/transforms_and_coordinate_systems/#rerun.DisconnectedSpace.__init__)
```

File: docs/content/reference/migration-0-9.md [hunk 3]
```diff
@@ -3,4 +3,238 @@ title: Migration to 0.9
...
+
+### `log_line_strip`, `log_line_strips_2d`, `log_line_strips_3d`, `log_line_segments`
+Replace with [LineStrips2D](data_types/archetypes/line_strips2d.md) or [LineStrips3D](data_types/archetypes/line_strips3d.md)
+
+Python docs: [LineStrips2D.__init__](https://ref.rerun.io/docs/python/HEAD/common/spatial_archetypes/#rerun.LineStrips2D.__init__), [LineStrips3D.__init__](https://ref.rerun.io/docs/python/HEAD/common/spatial_archetypes/#rerun.LineStrips3D.__init__)
+
+Notes:
+ - `log_line_segments` used to take an array of shape (2 * num_segments, 2 or 3) (where points were connected in
+even-odd pairs). Instead this is now handled by a batch of `LineStrips` all of length 2. Note that `LineStrips` now
+takes an array of shape (num_strips, num_points_per_strip, 2 or 3). You can use convert to the new format using the snippets:
```
Comment (emilk) [inline docs/content/reference/migration-0-9.md:118]: This makes it sounds like the array needs to be rectangular, i.e. that all strips in the same batch has the same number of points in it.

Rather, it takes a list of arrays, each of shape `(num_points_in_strip, 2 or 3)`

File: docs/content/reference/migration-0-9.md [hunk 4]
```diff
@@ -3,4 +3,238 @@ title: Migration to 0.9
...
+### `log_mesh_file`
+Replace with [Asset3D](data_types/archetypes/asset3d.md)
+
+Python docs: [Asset3D.__init__](https://ref.rerun.io/docs/python/HEAD/common/spatial_archetypes/#rerun.Asset3D.__init__)
+
+Notes:
+ - `mesh_bytes` and `mesh_path` are both now jut `data`. Strings and paths will be opened as files, while
+   file-descriptors or bytes objects will be read.
+ - `mesh_format` is now `media_type`.
+ - `transform` can now take anything that is compatible with `rr.Transform3D` instead of a 3x4 matrix.
```
Comment (emilk) [inline docs/content/reference/migration-0-9.md:149]: We should write out how to turn a 3x4 matrix into a `Transform3D`

File: docs/content/reference/migration-0-9.md [hunk 5]
```diff
@@ -3,4 +3,238 @@ title: Migration to 0.9
 order: 10
 ---
 
-[TODO(#3277)](https://github.com/rerun-io/rerun/issues/3277)
+Rerun-0.9 introduces a new set of object-oriented logging APIs built on top of an updated, more concrete,
```
Comment (nikolausWest) [inline docs/content/reference/migration-0-9.md:6]:
```suggestion
Rerun-0.9 introduces a new set of type-oriented logging APIs built on top of an updated, more concrete,
```

File: docs/content/reference/migration-0-9.md [hunk 6]
```diff
@@ -3,4 +3,238 @@ title: Migration to 0.9
...
+
+Python docs: [TextLog.__init__](https://ref.rerun.io/docs/python/HEAD/common/text/#rerun.TextLog.__init__)
+
+### `log_transform3d`
+Replace with [Transform3D](data_types/archetypes/transform3d.md)
+
+Python docs: [Transform3D.__init__](https://ref.rerun.io/docs/python/HEAD/common/transforms_and_coordinate_systems/#rerun.Transform3D.__init__)
+
+Notes:
+ - Now takes an optional parameters for `translation`, `rotation`, `scale`, or `mat3x3` to simplify construction.
```
Comment (emilk) [inline docs/content/reference/migration-0-9.md:231]:
```suggestion
 - Now takes optional parameters for `translation`, `rotation`, `scale`, or `mat3x3` to simplify construction.
```

**Comment by nikolausWest** [comment]
Nice!

File: rerun_py/docs/gen_common_index.py [hunk 7]
```diff
@@ -82,6 +82,12 @@ class Section:
         func_list=["log", "set_time_sequence", "set_time_seconds", "set_time_nanos"],
         class_list=[],
     ),
+    Section(
+        title="Adhoc Data",
```
Comment (emilk) [inline rerun_py/docs/gen_common_index.py:86]: "Any", "Adhoc", … we should have one coherent name for this. "Custom Data" is another. I don't know

**Review by emilk — APPROVED** [review]
This is excellent reading ⭐⭐⭐⭐⭐

File: docs/content/reference/migration-0-9.md (hunk 2 above)
Comment (jleibs) [reply docs/content/reference/migration-0-9.md:81]: Agreed.

File: docs/content/reference/migration-0-9.md (hunk 3 above)
Comment (jleibs) [reply docs/content/reference/migration-0-9.md:118]: Good catch.

File: rerun_py/docs/gen_common_index.py (hunk 7 above)
Comment (jleibs) [reply rerun_py/docs/gen_common_index.py:86]: I like Custom.

File: docs/content/reference/migration-0-9.md (hunk 4 above)
Comment (jleibs) [reply docs/content/reference/migration-0-9.md]: Would be even easier if we implement: https://github.com/rerun-io/rerun/issues/3645

**Pull request merged**